### PR TITLE
Add CONTRIBUTING.md file

### PR DESCRIPTION
**Pull request by gautamjajoo**
In continuation of #72, PR #93 only added COC.
@RishabhJain2018

### CODE REVIEW

File: .github/CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
+## Contributing guidelines
+
+Thank you for your interest in contributing to EvalAI! Here are a few pointers about how you can help.
```
Comment (Shekharrajak) [inline .github/CONTRIBUTING.md]: 'how you can help' -> 'how you can contribute'

File: .github/CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+### Setting things up
+
+To set up the development environment, follow the instructions in README.
```
Comment (Shekharrajak) [inline .github/CONTRIBUTING.md]: instrunction must contain the README#development setup section link.

File: .github/CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+### Finding something to work on
+
+The issue tracker of EvalAI a good place to start. If you find something that interests you, comment on the thread and we’ll help get you started.
+
+Alternatively, if you come across a new bug, please file a new issue and comment if you would like to be assigned. The existing issues are tagged with one or more labels, based on the part of the website it touches, its importance etc., that can help you in selecting one.
+
+If neither of these seem appealing, please post on our channel and we will help find you something else to work on.
```
Comment (Shekharrajak) [inline .github/CONTRIBUTING.md]: Gitter channel and mailing list link will be helpful for new people.

File: .github/CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+### Instructions to submit code
+
+Before you submit code, please talk to us via the issue tracker so we know you are working on it.
```
Comment (Shekharrajak) [inline .github/CONTRIBUTING.md]: 'so we know you are working on it' -> so we know you are working on it and also let us know your approach, to be sure that you are thinking on right direction.

File: .github/CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+
+Our central development branch is development. Coding is done on feature branches based off of development and merged into it once stable and reviewed. To submit code, follow these steps:
+
+1. Create a new branch off of development. Select a descriptive branch name.
```
Comment (Shekharrajak) [inline .github/CONTRIBUTING.md]: You can mention that below commands will help to get the latest version of the code from the upstream(master branch).

File: .github/CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+
+        git fetch upstream
+        git checkout master
+        git merge upstream/master
```
Comment (Shekharrajak) [inline .github/CONTRIBUTING.md]: `git merge` or `git rebase` either of one. `git rebase` for fast forwarding merge putting your commits up.

File: .github/CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+        git checkout -b your-branch-name
+
+2. Commit and push code to your branch:
+
+    - Commits should be self-contained and contain a descriptive commit message.
+    - Please make sure your code is well-formatted and adheres to PEP8 conventions (for Python) and the airbnb style guide (for JavaScript). For others (Lua, prototxt etc.) please ensure that the code is well-formatted and the style consistent.
```
Comment (Shekharrajak) [inline .github/CONTRIBUTING.md:49]: You can remove PEP8 here, sincve it coontains only frontend. basically they need to run `ng lint` command to resolve the linting issue if there is any. They can also add extension for Angular 7 in text editor to fix it while coding only.

File: .github/CONTRIBUTING.md [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+    - Please ensure that your code is well tested.
+    - We highly encourage to use `autopep8` to follow the PEP8 styling. Run the following command before creating the pull request:
```
Comment (Shekharrajak) [inline .github/CONTRIBUTING.md]: PEP8 not needed

File: .github/CONTRIBUTING.md [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+
+            autopep8 --in-place --exclude env,docs --recursive .     
+            git commit -a -m “{{commit_message}}”
+            git push origin {{branch_name}}
+    - Also, For Pretifying the Frontend Code Use ```HTML/JS/CSS Pretifier```.
+    - For installing the Sublime Package Control Manager in Sublime-Text Editor use [this](https://packagecontrol.io/installation#st2) link. Also, If Sublime Package Control Manager is installed then install ```HTML/JS/CSS Pretifier```.
+
+3. Once the code is pushed, create a pull request:
+
+    - On your Github fork, select your branch and click “New pull request”. Select “master” as the base branch and your branch in the “compare” dropdown.
```
Comment (Shekharrajak) [inline .github/CONTRIBUTING.md:64]: You can add github - contribution help link here.

**Comment by gautamjajoo** [comment]
@Shekharrajak This is the same file as it is in EvalAI repository. So do we need these changes?

**Comment by Shekharrajak** [comment]
@gautamjajoo , that's fine! But if my suggested comments are adding more clarity then we should go for it. Otherwise please let me know any strong point of not modifying those lines (like Python related).

**Comment by gautamjajoo** [comment]
@Shekharrajak Will make the required changes!

File: .github/CONTRIBUTING.md [hunk 10]
```diff
@@ -0,0 +1,49 @@
...
+
+If neither of these seem appealing, please post on our [gitter channel](https://gitter.im/Cloud-CV/EvalAI) and we will help find you something else to work on.
+
+### Instructions to submit code
+
+Before you submit code, please talk to us via the issue tracker so we know you are working on it and also let us know your approach, to be sure that you are thinking on right direction.
+
+Our central development branch is development. Coding is done on feature branches based off of development and merged into it once stable and reviewed. To submit code, follow these steps:
+
+1. Create a new branch off of development. Select a descriptive branch name. The commands below will help you to get the latest version of the code from the upstream(master branch).
```
Comment (Shekharrajak) [inline .github/CONTRIBUTING.md]: How to set the remote `upstream` ?

**Review by Shekharrajak — COMMENTED** [review]
Other than a comment, it is good to go :)

File: .github/CONTRIBUTING.md (hunk 10 above)
Comment (Shekharrajak) [reply .github/CONTRIBUTING.md]: You can add a note, something like:

If you have not added the upstream then run this command:

```
git remote add upstream https://github.com/Cloud-CV/EvalAI-ngx
```

and to verify : 

```
git remote -v
```

**Comment by RishabhJain2018** [comment]
@gautamjajoo Can you please incorporate @Shekharrajak's suggestion? Let's get this merged 🎉

**Comment by gautamjajoo** [comment]
@Shekharrajak Done the changes. Please review!

File: .github/CONTRIBUTING.md [hunk 11]
```diff
@@ -18,32 +18,54 @@ If neither of these seem appealing, please post on our [gitter channel](https://
...
-    - If your checks have passed, your PR will be assigned a reviewer who will review your code and provide comments. Please address each review comment by pushing new commits to the same branch (the PR will automatically update, so you don’t need to submit a new one). Once you are done, comment below each review comment marking it as “Done”. Feel free to use the thread to have a discussion about comments that you don’t understand completely or don’t agree with.
+    - On your Github fork, select your branch and click “New pull request”. Select “master” as the base branch and your branch in the         “compare” dropdown.
+      If the code is mergeable (you get a message saying “Able to merge”), go ahead and create the pull request.
+      
+    - Check back after some time to see if the Travis checks have passed, if not you should click on “Details” link on your PR thread at       the right of “The Travis CI build failed”, which will take you to the dashboard for your PR. You will see what failed / stalled,         and will need to resolve them.
```
Comment (Shekharrajak) [inline .github/CONTRIBUTING.md]: Not sure why there is many spaces between few words.

File: .github/CONTRIBUTING.md [hunk 12]
```diff
@@ -18,32 +18,54 @@ If neither of these seem appealing, please post on our [gitter channel](https://
...
     - Commits should be self-contained and contain a descriptive commit message.
-    - Please make sure your code is well-formatted and adheres to PEP8 conventions (for Python) and the airbnb style guide (for JavaScript). For others (Lua, prototxt etc.) please ensure that the code is well-formatted and the style consistent.
+    
+    - Please make sure your code is well-formatted and adheres to PEP8 conventions (for Python) and the airbnb style guide (for               JavaScript). For others (Lua, prototxt etc.) please ensure that the code is well-formatted and the style consistent.
+    
     - Please ensure that your code is well tested.
+    
     - If you have to check for any linting issues, run the following command before creating the pull request:
         ```ng lint```
+
```
Comment (Shekharrajak) [inline .github/CONTRIBUTING.md:57]: We can also write about running the testcases locally using `ng test` command

**Review by Shekharrajak — APPROVED** [review]
I suggested few things. It is good to go.

File: .github/CONTRIBUTING.md (hunk 12 above)
Comment (gautamjajoo) [reply .github/CONTRIBUTING.md:57]: Done.

**Comment by gautamjajoo** [comment]
Done the required changes!

**Comment by Shekharrajak** [comment]
LGTM :tada: .

**Pull request merged**